lib/puppetserver/ca/action/prune.rb

```diff
@@ -50,15 +50,24 @@ def run(inputs)
           loader = X509Loader.new(puppet.settings[:cacert], puppet.settings[:cakey], puppet.settings[:cacrl])
 
           puppet_crl = loader.crls.select { |crl| crl.verify(loader.key) }
-          prune_CRLs(puppet_crl)
-          update_pruned_CRL(puppet_crl, loader.key)
-          FileSystem.write_file(puppet.settings[:cacrl], loader.crls, 0644)
+          update_flag, dedupe_counter = prune_CRLs(puppet_crl)
+
+          if update_flag
```

Comment:
You should be able to get rid of the `update_flag` now and just check whether `dedupe_counter` is greater than 0.

Reply:
I also might call it something like `number_of_duplicate_revocations` or something.